test/tests/spiped-basics/run.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+[ "$DEBUG" ] && set -x
+
+set -eo pipefail
+
+dir="$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+image="$1"
+
+clientImage='buildpack-deps:buster-curl'
+# ensure the clientImage is ready and available
+if ! docker image inspect "$clientImage" &> /dev/null; then
+	docker pull "$clientImage" > /dev/null
+fi
+
+# Create an instance of the container-under-test
+serverImage="$("$dir/../image-name.sh" librarytest/spiped "$image")"
+"$dir/../docker-build.sh" "$dir" "$serverImage" <<EOD
+FROM $image
+COPY dir/keyfile /spiped/key
```

Comment:
Why commit an opaque keyfile when we could just generate one when running the test?

ala `RUN dd if=/dev/urandom bs=32 count=1 of=keyfile`

---

or maybe just `docker run ... sh -c "spiped-generate-key.sh && exec docker-entrypoint.sh -d -s ..."` with share for the key?

Reply:
I implemented it like that, because I remember from here that using the filesystem / bind mounts is unsafe in context of the tests: https://github.com/docker-library/official-images/pull/2097#issuecomment-244195736

But of course I should be able to replace the `COPY` with a `RUN` that statically bakes the keyfile into the created image.

Reply:
Yeah, we can't mount from the host, but `volumes-from` or a named volume with a `$RANDOM` element that gets cleaned up on `EXIT` should work fine.

Reply:
I've opted for `--volumes-from`.